Issue417 movers refactor

Mathadon opened the pull request:
For #417

mwetter commented on Annex60/Fluid/Movers/BaseClasses/FlowControlled.mo:
```diff
@@ -1,8 +1,31 @@
 within Annex60.Fluid.Movers.BaseClasses;
 partial model FlowControlled
   "Partial model for fan or pump with ideally controlled mass flow rate or head as input signal"
+  parameter Boolean use_record = false
```
Why using underscore when all other parameters use camel case?

Also, don't use pump, use movers as it is also used for fans

mwetter commented on Annex60/Fluid/Movers/BaseClasses/FlowMachineInterface.mo:
```diff
@@ -91,11 +107,19 @@ model FlowMachineInterface
 
  // Derivatives for cubic spline
 protected
+  final parameter Boolean preSpe = preVar == Annex60.Fluid.Types.PrescribedVariable.Speed
```
I think preSpe is only used once. Does it really make sense to introduce a protected parameter to use it one time only?

Mathadon replied:
For preSpe it may not be necessary, but for prePre it allows to add a comment and to provide some context there, so I'd prefer keeping it.

mwetter replied:
OK, sounds good.

mwetter commented on Annex60/Fluid/Movers/BaseClasses/FlowMachineInterface.mo:
```diff
@@ -469,112 +495,122 @@ the simulation stops.");
+                                                      delta=delta, d=preDer1, cBar=cBar, kRes=kRes)
+                                 +(V_flow-V_flow_nominal)*
+                                  (cha.pressure(per=pCur1,
+                                                      V_flow=(1+delta)*V_flow_nominal, r_N=1,
```
Can you make a smaller diff? I have a hard time checking what changed. For example, it used to be (1-delta) but now it is (1+delta)? This is probably a bug?

mwetter commented on Annex60/Fluid/Movers/Data/Pumps/Wilo/package.mo:
```diff
@@ -2,6 +2,7 @@ within Annex60.Fluid.Movers.Data.Pumps;
 package Wilo "Package with performance data for Wilo pumps"
   extends Modelica.Icons.Package;
 
+
```
No need for a change.

mwetter commented on Annex60/Fluid/Movers/Data/Pumps/package.mo:
```diff
@@ -2,6 +2,7 @@ within Annex60.Fluid.Movers.Data;
 package Pumps "Package containing data for pumps"
   extends Modelica.Icons.Package;
 
+
```
No need for a change.

mwetter commented on Annex60/Fluid/Movers/Data/SpeedControlled_y.mo:
```diff
@@ -12,8 +27,9 @@ record SpeedControlled_y
-  parameter Annex60.Fluid.Movers.BaseClasses.Characteristics.flowParameters pressure
-    "Volume flow rate vs. total pressure rise"
+  parameter Annex60.Fluid.Movers.BaseClasses.Characteristics.flowParameters pressure(
+    V_flow={0.5,1},
```
These default values must be removed. They are not meaningful in general.

mwetter commented on Annex60/Fluid/Movers/FlowControlled_dp.mo:
```diff
@@ -10,7 +10,8 @@ model FlowControlled_dp
      u(final unit="Pa"),
-     y(final unit="Pa")));
+     y(final unit="Pa")),
+    eff(preVar=Annex60.Fluid.Types.PrescribedVariable.PressureDifference));
```
Shouldn't that be `final` or under what circumstances is a change meaningful?

mwetter commented on Annex60/Fluid/Movers/FlowControlled_m_flow.mo:
```diff
@@ -10,7 +10,8 @@ model FlowControlled_m_flow
      u(final unit="kg/s"),
-     y(final unit="kg/s")));
+     y(final unit="kg/s")),
+    eff(preVar=Annex60.Fluid.Types.PrescribedVariable.FlowRate));
```
final?

Mathadon commented:
@mwetter I addressed all concerns
I did not update reference results so you can verify
some models now have more algebraic loops since the exact power computation is not used by default, since it is automatically activated when the required data is present.
